### DISCUSSION TITLE
Converting WPF application to MAUI

### DISCUSSION THREAD

**Discussion by slnsvr** [dotnet/maui#3905] [Q&A]
We have information that Xamarin.Forms applications can be upgraded to MAUI. But in addition to that will we have an option to convert WPF applications also to .NET MAUI? Asking this because we want to make our WPF application to make cross platform. And if its possible to migrate from WPF to MAUI, is there is any tool for doing the same? If migration is possible and if tool is not available right now, are there any plans in future to make the tool available?

**Comment by chenjing1294** [comment]
i want to know too

**Comment by MissedSte4k** [comment]
Don't know about tools, but can answer from my own experience. I started developing on Maui 
about a year ago. Came from WPF background. Migrating the ViewModels/Views from WPF to Maui by hand was more or less peace of cake (need to mention that it was more of PC -> Android migration for me, so a lot of minor UI changes on the way). You just need to figure out what corresponds to what and then everything is good. The main problem is: Maui still has way too many bugs for it to be of any use to migrate into at this point. I've been developing multiple applications for about a year now. And the amount of stalemates I went into during this time is insane. I'm creating sample apps for our software/API. So at this point - when I get into stalemate (bugs that completely stop development until they are fixed) I just go start/switch to develop another app. At this point - I have 2 95% ready to launch apps that I can't send to testing team or launch, because of MAUI issues. At the moment - I think I'm following around 10-11 different issues. The count of issues for me is just growing. Issues are not solved and are just piling up. Those who get solved - takes ages to be pushed. That's why there are over 1.6k unsolved issues atm. 

Some sneak peak of what you will find:
Android crash on release version if ZIndex is used: https://github.com/dotnet/maui/issues/7331
Wrong images being displayed: https://github.com/dotnet/maui/issues/8676
Crashes due to image cashing: https://github.com/dotnet/maui/issues/9011
ListView is more/less unusable: https://github.com/dotnet/maui/issues/5292
iOS icons are not scaled properly: https://github.com/dotnet/maui/issues/8609
Collection in iOS and Windows are not scaled correctly: https://github.com/dotnet/maui/issues/8636
Random ItemTemplates: https://github.com/dotnet/maui/issues/8811
SelectedItem not workeing if Frame is used: https://github.com/dotnet/maui/issues/7121

I'm starting to think, that they'll just gonna drop this project all together sometimes :(  
I would suggest - If you didn't start the migration - save your sanity and don't do it at least for now. And if you did - may the god have mercy on your soul and equip yourself with a lot of patients. You will need it.

**Reply by yevgeny-sotnikov** [reply]
My experience of migration - MAUI doesn't support printing at the same as WPF does: [#9931](https://github.com/dotnet/maui/issues/9931)
